File: src/main/java/com/lp/v2/domains/auth/RefreshTokenEntity.java
package com.lp.v2.domains.auth;

import com.lp.v2.common.Auditable;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class RefreshTokenEntity extends Auditable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "token", length = 512, unique = true, nullable = false)
    private String token;

    @Column(name = "account_id", nullable = false)
    private Long accountId;

    @Column(name = "issued_at", nullable = false)
    private LocalDateTime issuedAt;

    @Column(name = "expires_at", nullable = false)
    private LocalDateTime expiresAt;

}

File: lp-storage/src/main/kotlin/com/boaz/lp/storage/entity/RefreshTokenEntity.kt
package com.boaz.lp.storage.entity

import com.boaz.lp.storage.entity.base.BaseEntity
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(
    name = "refresh_token",
    indexes = [
        Index(name = "idx_refresh_token_member_id", columnList = "member_id"),
        Index(name = "idx_refresh_token_token", columnList = "token")
    ]
)
class RefreshTokenEntity(

    @Column(name = "member_id", nullable = false)
    var memberId: Long,

    @Column(name = "token", nullable = false, unique = true, length = 512)
    var token: String,

    @Column(name = "expires_at", nullable = false)
    var expiresAt: LocalDateTime

) : BaseEntity() {

    fun isExpired(): Boolean = LocalDateTime.now().isAfter(expiresAt)
}
